tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs

```diff
@@ -614,6 +614,8 @@ private void AddOrRemoveServers(ClusterDescription clusterDescription)
 
     internal static class ClusterReflector
     {
+        public static void _clusterId(this Cluster cluster, ClusterId clusterId) => Reflector.SetFieldValue(cluster, nameof(_clusterId), clusterId);
+        public static void _description(this Cluster cluster, ClusterDescription clusterDescription) => Reflector.SetFieldValue(cluster, nameof(_description), clusterDescription);
```

Comment:
Remove these two lines from `ClusterReflector`. Instead put this logic in a helper method the CSharp3173Tets class:

    private void ForceClusterId(MultiServerCluster cluster, ClusterId clusterId)
    {
        Reflector.SetFieldValue(cluster, "_clusterId", clusterId);
        Reflector.SetFieldValue(cluster, "_description", ClusterDescription.CreateInitial(__clusterId, __clusterConnectionMode));
    }


I don't want to encourage any other tests from overwriting these fields.

Reply:
done